### ISSUE TITLE
it's not possible to run the app with flutter run or distribute it in the store ios

### ISSUE THREAD

**Issue by insinfo** [flutter/flutter#93192]
it's not possible to run the app with flutter run or distribute it in the store, I'm using ios 15.1, macos 12.0.1 and xcode 13.1 to publish an update to an app, but when I try to run the app with flutter run I get these errors.

```console
macbook:appjazzmobile insinfo$ flutter run
Launching lib/main.dart on iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: 26JLPAYSU4
Running Xcode build...                                                  
Xcode build done.                                            6,4s
Failed to build iOS app
Error output from Xcode build:
↳
    2021-11-06 13:16:35.676 xcodebuild[22651:161883]  DVTAssertions: Warning in
    /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-19450/DTDeviceKitBase/DTDKRemoteDeviceData.m:373
    Details:  (null) deviceType from 7501aa1d65033b44c4eaa61dce40b64699bd96cc was NULL when -platform called.
    Object:   <DTDKMobileDeviceToken: 0x7fafb6260d80>
    Method:   -platform
    Thread:   <NSThread: 0x600003a4c400>{number = 4, name = (null)}
    Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
    2021-11-06 13:16:35.879 xcodebuild[22651:161882]  DVTAssertions: Warning in
    /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-19450/DTDeviceKitBase/DTDKRemoteDeviceData.m:373
    Details:  (null) deviceType from 7501aa1d65033b44c4eaa61dce40b64699bd96cc was NULL when -platform called.
    Object:   <DTDKMobileDeviceToken: 0x7fafb6260d80>
    Method:   -platform
    Thread:   <NSThread: 0x600003578440>{number = 2, name = (null)}
    Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
    2021-11-06 13:16:35.947 xcodebuild[22651:161882]  DVTAssertions: Warning in
    /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-19450/DTDeviceKitBase/DTDKRemoteDeviceData.m:373
    Details:  (null) deviceType from 7501aa1d65033b44c4eaa61dce40b64699bd96cc was NULL when -platform called.
    Object:   <DTDKMobileDeviceToken: 0x7fafb6260d80>
    Method:   -platform
    Thread:   <NSThread: 0x600003578440>{number = 2, name = (null)}
    Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
    --- xcodebuild: WARNING: Using the first of multiple matching destinations:
    { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
    { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
    { platform:iOS Simulator, id:B5AF28CC-DA34-49B6-90ED-D841BFA55326, OS:15.0, name:iPad (9th generation) }
    { platform:iOS Simulator, id:F6D5530A-717C-4DAD-9018-1B211B9F76C4, OS:15.0, name:iPad Air (4th generation) }
    { platform:iOS Simulator, id:DE8B0798-D6E4-44F8-9FA5-D2375866FCFD, OS:15.0, name:iPad Pro (9.7-inch) }
    { platform:iOS Simulator, id:264EC260-E1E8-41A6-B903-11AB352B013B, OS:15.0, name:iPad Pro (11-inch) (3rd generation) }
    { platform:iOS Simulator, id:9AD008DA-7D09-4788-A0EB-D677AB6286C7, OS:15.0, name:iPad Pro (12.9-inch) (5th generation) }
    { platform:iOS Simulator, id:3FD493E7-B086-452B-8198-41023096AA9F, OS:15.0, name:iPad mini (6th generation) }
    { platform:iOS Simulator, id:53B8334F-B5E7-44E5-840E-FDF334FBD089, OS:15.0, name:iPhone 8 }
    { platform:iOS Simulator, id:8D5F96FE-529E-49CB-96BB-0BEDA6DB3213, OS:15.0, name:iPhone 8 Plus }
    { platform:iOS Simulator, id:3CBA5525-D87C-454D-98B7-A415BE786FFD, OS:15.0, name:iPhone 11 }
    { platform:iOS Simulator, id:052D5560-F0C6-4C8A-8780-3D4284AAB47F, OS:15.0, name:iPhone 11 Pro }
    { platform:iOS Simulator, id:12AD14F2-6861-4712-AB4C-3FFA42062C5D, OS:15.0, name:iPhone 11 Pro Max }
    { platform:iOS Simulator, id:39312C9E-00DA-4BDA-B9C0-142F9AF76ECB, OS:15.0, name:iPhone 12 }
    { platform:iOS Simulator, id:D8D5998B-8D8D-47D9-8E86-4E7F6C55A98A, OS:15.0, name:iPhone 12 Pro }
    { platform:iOS Simulator, id:D496480A-F195-4B55-87B8-DB54065D2D2F, OS:15.0, name:iPhone 12 Pro Max }
    { platform:iOS Simulator, id:E0CC74E5-824A-4CE9-A4DE-A3B048AF044C, OS:15.0, name:iPhone 12 mini }
    { platform:iOS Simulator, id:840EAEA5-6613-4354-A4B5-8881FD370C56, OS:15.0, name:iPhone 13 }
    { platform:iOS Simulator, id:DC20EBED-8149-4930-8D4A-D64E3E89FE65, OS:15.0, name:iPhone 13 Pro }
    { platform:iOS Simulator, id:093559FE-39DB-4786-9714-89873B900BCE, OS:15.0, name:iPhone 13 Pro Max }
    { platform:iOS Simulator, id:29B791A4-0079-44D3-9E3A-BCFFEAC0FCA8, OS:15.0, name:iPhone 13 mini }
    { platform:iOS Simulator, id:57226ECC-D828-4F51-A56E-69F62E741AB1, OS:15.0, name:iPhone SE (2nd generation) }
    { platform:iOS Simulator, id:1E714389-D228-4012-AF6C-5A14A424519C, OS:15.0, name:iPod touch (7th generation) }
    ** BUILD FAILED **


Xcode's output:
↳
    While building module 'GoogleUtilities' imported from
    /Users/insinfo/Desktop/appjazzmobile/ios/Pods/FirebaseCoreDiagnostics/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:22:
    In file included from <module-includes>:1:
    /Users/insinfo/Desktop/appjazzmobile/ios/Pods/Target Support Files/GoogleUtilities/GoogleUtilities-umbrella.h:39:1: error: umbrella header for
    module 'GoogleUtilities' does not include header 'GULSwizzler.h' [-Werror,-Wincomplete-umbrella]
    /Users/insinfo/Desktop/appjazzmobile/ios/Pods/Target Support Files/GoogleUtilities/GoogleUtilities-umbrella.h:39:1: error: umbrella header for
    module 'GoogleUtilities' does not include header 'GULUserDefaults.h' [-Werror,-Wincomplete-umbrella]
    /Users/insinfo/Desktop/appjazzmobile/ios/Pods/Target Support Files/GoogleUtilities/GoogleUtilities-umbrella.h:39:1: error: umbrella header for
    module 'GoogleUtilities' does not include header 'GULOriginalIMPConvenienceMacros.h' [-Werror,-Wincomplete-umbrella]
    3 errors generated.
    /Users/insinfo/Desktop/appjazzmobile/ios/Pods/FirebaseCoreDiagnostics/Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m:22:9: fatal
    error: could not build module 'GoogleUtilities'
    #import <GoogleUtilities/GULAppEnvironmentUtil.h>
     ~~~~~~~^
    4 errors generated.
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel

Could not build the precompiled application for the device.

Error launching application on iPhone.
macbook:appjazzmobile insinfo$ 
```

when I compile with xcode in the project it runs normally on the iphone, but when I try to distribute the app to the store, the error appears

```
invalid bundle. the bundle runner.app/Frameworks/frutter.framework does not support the minimum version
```

![WhatsApp Image 2021-11-06 at 13 39 13](https://user-images.githubusercontent.com/12227024/140617152-b9cbeb9c-8c55-4a09-979c-f21701747599.jpeg)

### flutter doctor -v
```console
macbook:appjazzmobile insinfo$ flutter doctor -v
[✓] Flutter (Channel dev, 2.6.0-11.0.pre, on macOS 12.0.1 21A559 darwin-x64, locale pt-BR)
    • Flutter version 2.6.0-11.0.pre at /Users/insinfo/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4b330ddbed (7 weeks ago), 2021-09-16 17:29:58 -0700
    • Engine revision 5b81c6d615
    • Dart version 2.15.0 (build 2.15.0-116.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/insinfo/Library/Android/sdk
    • Platform android-31, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (3 available)
    • iPhone (mobile) • 7501aa1d65033b44c4eaa61dce40b64699bd96cc • ios            • iOS 15.1 19B74
    • macOS (desktop) • macos                                    • darwin-x64     • macOS 12.0.1 21A559 darwin-x64
    • Chrome (web)    • chrome                                   • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```
### Podfile
```Podfile
# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!
  #https://firebase.flutter.dev/docs/overview#initializing-flutterfire
  pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '8.9.1'

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end

```

### pubspec.yaml
```yaml
name: appjazz
description: Aplicativo do Jazz e Blues
version: 9.0.0+12
publish_to: none

environment:
  sdk: ">=2.10.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  #cupertino_icons: ^1.0.3 #Null safety

  http: ^0.13.4 #Null safety
  path_provider: ^2.0.5 #Null safety  
  sqflite: ^2.0.0+4 #Null safety
  connectivity: ^3.0.6  #Null safety sera descontinuado final 2021 usar este connectivity_plus 2.0.2 no lugar
  url_launcher: ^6.0.12 #Null safety

  firebase_core: ^1.8.0 #Null safety
  cloud_firestore: ^2.5.4 #Null safety
  firebase_auth: ^3.1.4 #Null safety
  #google_sign_in: ^5.2.0 #Null safety 
  #firebase_analytics: ^8.3.4 #Null safety 

  rxdart: ^0.27.2 #Null safety   
  webview_flutter: ^2.1.2 #Null safety 
  auto_size_text: ^3.0.0-nullsafety.0 #Null safety 
  flutter_map: ^0.14.0 #Null safety 
  shared_preferences: ^2.0.8 #Null safety
  latlong2: ^0.8.1 #Null safety
```

**Comment by darshankawar** [comment]
@insinfo 
This was fixed and merged into master, https://github.com/flutter/flutter/issues/90209 but may not be in dev yet (https://github.com/flutter/flutter/issues/90209#issuecomment-939435041), so, you may need to switch to latest master to see it working.
I am going ahead and closing it as already fixed. If after switching to master, you see the same issue, write in comments here and I'll reopen.

**Comment by SomeoneAndNoone** [comment]
@darshankawar when will this go to release in the next version of Flutter?
